tfprotov5/tftypes/value_test.go

```diff
@@ -58,19 +88,130 @@ func TestValueAs(t *testing.T) {
+			as:       numberPointerPointer(big.NewFloat(123)),
+			expected: numberPointerPointer(nil),
+		},
+		"number-pointer-int16": {
+			in:       NewValue(Number, int16Pointer(123)),
+			as:       big.NewFloat(0),
+			expected: big.NewFloat(123),
+		},
+		"number-pointer-int16-null": {
+			in:       NewValue(Number, (*int16)(nil)),
+			as:       numberPointerPointer(big.NewFloat(123)),
+			expected: numberPointerPointer(nil),
+		},
+		"number-pointer-int8": {
+			in:       NewValue(Number, int8Pointer(123)),
+			as:       big.NewFloat(0),
+			expected: big.NewFloat(123),
+		},
+		"number-pointer-int8-null": {
+			in:       NewValue(Number, (*int8)(nil)),
+			as:       numberPointerPointer(big.NewFloat(123)),
+			expected: numberPointerPointer(nil),
+		},
+		"number-pointer-int": {
+			in:       NewValue(Number, intPointer(123)),
+			as:       big.NewFloat(0),
+			expected: big.NewFloat(123),
+		},
+		"number-pointer-int-null": {
+			in:       NewValue(Number, (*int)(nil)),
+			as:       numberPointerPointer(big.NewFloat(123)),
+			expected: numberPointerPointer(nil),
+		},
+		"number-pointer-uint64": {
+			in:       NewValue(Number, uint64Pointer(123)),
+			as:       big.NewFloat(0),
+			expected: big.NewFloat(123),
+		},
+		"number-pointer-uint64-null": {
+			in:       NewValue(Number, (*uint64)(nil)),
+			as:       numberPointerPointer(big.NewFloat(123)),
+			expected: numberPointerPointer(nil),
+		},
+		"number-pointer-uint32": {
+			in:       NewValue(Number, uint32Pointer(123)),
+			as:       big.NewFloat(0),
+			expected: big.NewFloat(123),
+		},
+		"number-pointer-uint32-null": {
+			in:       NewValue(Number, (*uint32)(nil)),
+			as:       numberPointerPointer(big.NewFloat(123)),
+			expected: numberPointerPointer(nil),
+		},
+		"number-pointer-uint16": {
+			in:       NewValue(Number, uint16Pointer(123)),
+			as:       big.NewFloat(0),
+			expected: big.NewFloat(123),
+		},
+		"number-pointer-uint16-null": {
+			in:       NewValue(Number, (*uint16)(nil)),
+			as:       numberPointerPointer(big.NewFloat(123)),
+			expected: numberPointerPointer(nil),
+		},
+		"number-pointer-uint8": {
+			in:       NewValue(Number, uint8Pointer(123)),
+			as:       big.NewFloat(0),
+			expected: big.NewFloat(123),
+		},
+		"number-pointer-uint8-null": {
+			in:       NewValue(Number, (*uint8)(nil)),
+			as:       numberPointerPointer(big.NewFloat(123)),
+			expected: numberPointerPointer(nil),
+		},
+		"number-pointer-uint": {
+			in:       NewValue(Number, uintPointer(123)),
+			as:       big.NewFloat(0),
+			expected: big.NewFloat(123),
+		},
+		"number-pointer-uint-null": {
+			in:       NewValue(Number, (*uint)(nil)),
+			as:       numberPointerPointer(big.NewFloat(123)),
+			expected: numberPointerPointer(nil),
+		},
 		"number-null": {
-			in:       NewValue(Number, nil),
-			as:       big.NewFloat(123),
+			in: NewValue(Number, nil),
+			as: big.NewFloat(123),
+			// Shouldn't this be nil? unsure why we can't set that value?
```

Comment:
@paddycarver left this question here, curious why you shouldn't expect a nil here for the big float?

Reply:
We can drop this comment if its intentional, just seemed weird to me, not what I would have expected if you have a nil-able type.

```suggestion
```

Reply:
I wanted the behavior to be consistent--one pointer, zero value, double pointer, nil. `big.Float` is weird because it's _normally_ passed around boy pointer, but I felt it was more consistent.